File: src/main/java/xyz/devvydont/smprpg/items/tools/ItemHatchet.java
package xyz.devvydont.smprpg.items.tools;

import org.bukkit.inventory.EquipmentSlotGroup;
import org.bukkit.inventory.ItemStack;
import xyz.devvydont.smprpg.attribute.AttributeWrapper;
import xyz.devvydont.smprpg.items.CustomItemType;
import xyz.devvydont.smprpg.items.ItemClassification;
import xyz.devvydont.smprpg.items.attribute.AdditiveAttributeEntry;
import xyz.devvydont.smprpg.items.attribute.AttributeEntry;
import xyz.devvydont.smprpg.items.attribute.MultiplicativeAttributeEntry;
import xyz.devvydont.smprpg.items.base.CustomAttributeItem;
import xyz.devvydont.smprpg.items.interfaces.IBreakableEquipment;
import xyz.devvydont.smprpg.services.ItemService;

import java.util.Collection;
import java.util.List;

public class ItemHatchet extends CustomAttributeItem implements IBreakableEquipment {

    public double getHatchetMiningPower() {
        return 0;
    }

    public double getHatchetDamage() {
        return 1;
    }

    public double getHatchetFortune() {
        return 1;
    }

    public double getHatchetSpeed() {
        return 1;
    }

    public int getPowerRating() {
        return 0;
    }

    public static double HATCHET_ATTACK_SPEED_DEBUFF = -0.8;

    public ItemHatchet(ItemService itemService, CustomItemType type) {
        super(itemService, type);
    }

    @Override
    public ItemClassification getItemClassification() {
        return ItemClassification.HATCHET;
    }

    @Override
    public Collection<AttributeEntry> getAttributeModifiers(ItemStack item) {
        return List.of(
                new AdditiveAttributeEntry(AttributeWrapper.MINING_POWER, getHatchetMiningPower()),
                new AdditiveAttributeEntry(AttributeWrapper.STRENGTH, getHatchetDamage()),
                new MultiplicativeAttributeEntry(AttributeWrapper.ATTACK_SPEED, HATCHET_ATTACK_SPEED_DEBUFF),
                new AdditiveAttributeEntry(AttributeWrapper.MINING_SPEED, getHatchetSpeed()),
                new AdditiveAttributeEntry(AttributeWrapper.WOODCUTTING_FORTUNE, getHatchetFortune()),
                new AdditiveAttributeEntry(AttributeWrapper.FARMING_FORTUNE, getHatchetFortune())
        );
    }

    @Override
    public EquipmentSlotGroup getActiveSlot() {
        return EquipmentSlotGroup.MAINHAND;
    }

    @Override
    public int getMaxDurability() {
        return 50_000;
    }
}

File: src/main/java/xyz/devvydont/smprpg/items/tools/ItemHatchet.kt
package xyz.devvydont.smprpg.items.tools

import org.bukkit.inventory.EquipmentSlotGroup
import org.bukkit.inventory.ItemStack
import xyz.devvydont.smprpg.attribute.AttributeWrapper
import xyz.devvydont.smprpg.items.CustomItemType
import xyz.devvydont.smprpg.items.ItemClassification
import xyz.devvydont.smprpg.items.attribute.AdditiveAttributeEntry
import xyz.devvydont.smprpg.items.attribute.AttributeEntry
import xyz.devvydont.smprpg.items.attribute.MultiplicativeAttributeEntry
import xyz.devvydont.smprpg.items.base.CustomAttributeItem
import xyz.devvydont.smprpg.items.interfaces.IBreakableEquipment
import xyz.devvydont.smprpg.items.interfaces.IDamageFromCrops
import xyz.devvydont.smprpg.services.ItemService

open class ItemHatchet(itemService: ItemService, type: CustomItemType) : CustomAttributeItem(itemService, type),
    IBreakableEquipment, IDamageFromCrops {

    override val itemClassification: ItemClassification get() = ItemClassification.HATCHET

    open val hatchetMiningPower: Double get() = 0.0
    open val hatchetDamage: Double get() = 1.0
    open val hatchetFortune: Double get() = 1.0
    open val hatchetSpeed: Double get() = 1.0

    override fun getPowerRating(): Int { return 0 }

    override fun getAttributeModifiers(item: ItemStack?): MutableCollection<AttributeEntry?> {
        return mutableListOf(
            AdditiveAttributeEntry(AttributeWrapper.MINING_POWER, this.hatchetMiningPower),
            AdditiveAttributeEntry(AttributeWrapper.STRENGTH, this.hatchetDamage),
            MultiplicativeAttributeEntry(AttributeWrapper.ATTACK_SPEED, HATCHET_ATTACK_SPEED_DEBUFF),
            AdditiveAttributeEntry(AttributeWrapper.MINING_SPEED, this.hatchetSpeed),
            AdditiveAttributeEntry(AttributeWrapper.WOODCUTTING_FORTUNE, this.hatchetFortune),
            AdditiveAttributeEntry(AttributeWrapper.FARMING_FORTUNE, this.hatchetFortune)
        )
    }

    override fun getActiveSlot(): EquipmentSlotGroup {
        return EquipmentSlotGroup.MAINHAND
    }

    override fun getMaxDurability(): Int {
        return 50000
    }

    companion object {
        var HATCHET_ATTACK_SPEED_DEBUFF: Double = -0.8
    }
}
